### PR TITLE
Update and expand SimpleSAML for Drupal documentation

### PR DESCRIPTION
**Pull request by Crell**
After some testing with a customer, I present this revised and so-far-working SimpleSAMLphp documentation for both Drupal 7 and Drupal 8.

### CODE REVIEW

File: src/frameworks/drupal7/simplesaml.md [hunk 1]
```diff
@@ -96,6 +101,22 @@ if (isset($_ENV['PLATFORM_PROJECT_ENTROPY'])) {
 }
 ```
 
+## Generate SSL certs (optional)
+
+Depending on your SimpleSAML Service Provider (SP), you may need to generate an SSL/TLS certificate to connect to the SP.  If so, you should generate the cert locally following the instructions in the [SimpleSAMLphp documentation](https://simplesamlphp.org/docs/stable/simplesamlphp-sp).  Whatever your resulting idP file is should be placed in the simplesamlphp/metadata directory.  The certificate should be placed in the `simplesamlphp/cert` directory.  (Create it if needed.)
```
Comment (pjcdawkins) [inline src/frameworks/drupal7/simplesaml.md]: I think this should be "Depending on your Identity Provider (IdP)..." and "...connect to the IdP". The SP is the thing we are currently configuring.

File: src/frameworks/drupal8/simplesaml.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+// Set the salt value from the Platform.sh entropy value, provided for this purpose.
+if (isset($_ENV['PLATFORM_PROJECT_ENTROPY'])) {
+  $config['secretsalt'] = $_ENV['PLATFORM_PROJECT_ENTROPY'];
+}
+```
+
+## Generate SSL certs (optional)
+
+Depending on your SimpleSAML Service Provider (SP), you may need to generate an SSL/TLS certificate to connect to the SP.  If so, you should generate the cert locally following the instructions in the [SimpleSAMLphp documentation](https://simplesamlphp.org/docs/stable/simplesamlphp-sp).  Whatever your resulting idP file is should be placed in the simplesamlphp/metadata directory.  The certificate should be placed in the `simplesamlphp/cert` directory.  (Create it if needed.)
```
Comment (pjcdawkins) [inline src/frameworks/drupal8/simplesaml.md]: See https://github.com/platformsh/platformsh-docs/pull/648/files#r144503729

File: src/frameworks/drupal7/simplesaml.md [hunk 3]
```diff
@@ -96,6 +101,22 @@ if (isset($_ENV['PLATFORM_PROJECT_ENTROPY'])) {
 }
 ```
 
+## Generate SSL certs (optional)
+
```
Comment (ralt) [inline src/frameworks/drupal7/simplesaml.md]: What's SP?

Comment (ralt) [reply src/frameworks/drupal7/simplesaml.md]: > generate the cert

Let's use the full word?

Comment (ralt) [reply src/frameworks/drupal7/simplesaml.md]: "is should"?

File: src/frameworks/drupal7/simplesaml.md [hunk 4]
```diff
@@ -96,6 +101,22 @@ if (isset($_ENV['PLATFORM_PROJECT_ENTROPY'])) {
...
+Depending on your Identity Provider (IdP), you may need to generate an SSL/TLS certificate to connect to the SP.  If so, you should generate the cert locally following the instructions in the [SimpleSAMLphp documentation](https://simplesamlphp.org/docs/stable/simplesamlphp-sp).  Whatever your resulting idP file is should be placed in the simplesamlphp/metadata directory.  The certificate should be placed in the `simplesamlphp/cert` directory.  (Create it if needed.)
+
+Then add the following line to your `simplesamlphp/config/config.php` file to tell the library where to find the cert:
```
Comment (ralt) [inline src/frameworks/drupal7/simplesaml.md]: Same here, "cert"

File: src/frameworks/drupal7/simplesaml.md [hunk 5]
```diff
@@ -96,6 +101,22 @@ if (isset($_ENV['PLATFORM_PROJECT_ENTROPY'])) {
...
+
+```php
+$config['certdir'] = dirname(__DIR__) . '/cert';
```
Comment (ralt) [inline src/frameworks/drupal7/simplesaml.md]: `dirname(__DIR__)`?

File: src/frameworks/drupal7/simplesaml.md [hunk 6]
```diff
@@ -96,6 +101,22 @@ if (isset($_ENV['PLATFORM_PROJECT_ENTROPY'])) {
...
+```
+
+## Deploy
+
+Commit all changes and deploy the site, then enable the `simplesamlphp_auth` module within Drupal.
```
Comment (ralt) [inline src/frameworks/drupal7/simplesaml.md]: Or do that in a hook_update, triggered by the deploy hook?

File: src/frameworks/drupal8/simplesaml.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    locations:
+        '/simplesaml':
+            root: 'vendor/simplesamlphp/simplesamlphp/www'
+            allow: true
+            scripts: true
+            index:
+                - index.php
+```
+
+That will map all requests to `example.com/simplesaml/` to the `simplesamlphp/www` directory, allowing static files there to be served, PHP scripts to execute, and defaulting to index.php.
```
Comment (ralt) [inline src/frameworks/drupal8/simplesaml.md]: Put the full path to the directory?

File: src/frameworks/drupal8/simplesaml.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+## Create a configuration directory
+
+Your SimpleSAMLphp configuration will need to be outside of the `vendor` directory.  The `composer require` will download a template configuration file to `vendor/simplesamlphp/simplesamlphp/config`.
+
+Rather than modifying that file in place (as it won't be included in Git), copy the `vendor/simplesamlphp/simplesamlphp/config` directory to `simplesamlphp/config` (in your application root).  It should contain two files, `config.php` and `authsources.php`.
+
+Additionally, create a `simplesamplephp/metadata` directory.  This directory will hold your idP definitions.  Consult the SimpleSAMLphp documentation and see the examples in `vendor/simplesamplephp/simplesamlphp/metadata-templates`.
```
Comment (ralt) [inline src/frameworks/drupal8/simplesaml.md]: "IdP", not "idP"

File: src/frameworks/drupal8/simplesaml.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+// Set the salt value from the Platform.sh entropy value, provided for this purpose.
+if (isset($_ENV['PLATFORM_PROJECT_ENTROPY'])) {
+  $config['secretsalt'] = $_ENV['PLATFORM_PROJECT_ENTROPY'];
+}
+```
+
+## Generate SSL certs (optional)
+
```
Comment (ralt) [inline src/frameworks/drupal8/simplesaml.md]: "certificates"?

File: src/frameworks/drupal8/simplesaml.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+Depending on your Identity Provider (IdP), you may need to generate an SSL/TLS certificate to connect to the SP.  If so, you should generate the cert locally following the instructions in the [SimpleSAMLphp documentation](https://simplesamlphp.org/docs/stable/simplesamlphp-sp).  Whatever your resulting idP file is should be placed in the simplesamlphp/metadata directory.  The certificate should be placed in the `simplesamlphp/cert` directory.  (Create it if needed.)
```
Comment (ralt) [inline src/frameworks/drupal8/simplesaml.md]: SP?

File: src/frameworks/drupal7/simplesaml.md (hunk 3 above)
Comment (Crell) [reply src/frameworks/drupal7/simplesaml.md]: "Whatever the file is" / "should be placed in".  It works grammatically, although I'm open to alternatives.

File: src/frameworks/drupal7/simplesaml.md (hunk 5 above)
Comment (Crell) [reply src/frameworks/drupal7/simplesaml.md]: Shortest way to get the parent directory.  dirname() takes a path and trims off the last part, effectively giving you `__DIR__ . '/..'`.  (I didn't realize it either, but the customer I was working with on these docs found it worked, and it's nice and short!)

File: src/frameworks/drupal7/simplesaml.md (hunk 3 above)
Comment (pjcdawkins) [reply src/frameworks/drupal7/simplesaml.md]: Resolved

**Pull request merged**